src/main/java/io/tackle/pathfinder/services/AssessmentSvc.java

```diff
@@ -118,9 +114,76 @@ private Questionnaire defaultQuestionnaire() {
 
     public AssessmentDto getAssessmentDtoByAssessmentId(@NotNull Long assessmentId) {
         log.log(Level.FINE,"Requesting Assessment " + assessmentId);
-        Assessment assessment = (Assessment) Assessment.findByIdOptional(assessmentId).orElseThrow(() -> new NotFoundException());
+        Assessment assessment = (Assessment) Assessment.findByIdOptional(assessmentId).orElseThrow(NotFoundException::new);
 
         return mapper.assessmentToAssessmentDto(assessment);
     }
 
+    @Transactional
+    public AssessmentHeaderDto updateAssessment(@NotNull Long assessmentId, @NotNull @Valid AssessmentDto assessmentDto) {
+        Assessment assessment = (Assessment) Assessment.findByIdOptional(assessmentId).orElseThrow(NotFoundException::new);
+        if (null != assessmentDto.getStakeholderGroups()) {
+            // Delete existing stakeholdergroups not included in current array
+            assessment.stakeholdergroups.forEach(stakegroup -> {
+                if (!assessmentDto.getStakeholderGroups().contains(stakegroup.stakeholdergroupId)) {
+                    log.info("Deleted stakegroup : " + stakegroup.stakeholdergroupId);
+                    stakegroup.delete();
+                }
+            });
+            // Add not existing stakeholdergroups included in the current array
+            assessmentDto.getStakeholderGroups().forEach(e -> {
+                log.info("Considering Stakeholdergroup : " + e);
```

Comment:
this method's log calls would be better as debug level

Reply:
:+1: